NetJSON/NetJSON.cs

```diff
@@ -3727,80 +3727,87 @@ public static object Deserialize(Type type, string value, NetJSONSettings settin
             {
                 return null;
             }
+			ptr += index;
+			var startPtr = ptr;
 
-            while (true) {
-                current = ptr[index];
-                if(current == '\0')
-                {
-                    break;
-                }
+            while (*ptr != '\0') {
+                current = *ptr;
 
                 if (isJustString || hasQuote) {
                     if (!isJustString && current == settings._quoteChar)
                     {
-                        next = ptr[index + 1];
+                        next = *(++ptr);
```

Comment:
Nice. you read the next piece of data knowing that it will either exit the loop or throw exception rather than incrementing wasting "index + 1"